src/backend/distributed/deparser/deparse_role_stmts.c

```diff
@@ -0,0 +1,139 @@
+/*-------------------------------------------------------------------------
+ *
+ * deparse_role_stmts.c
+ *	  All routines to deparse role statements.
+ *	  This file contains all entry points specific for ALTER ROLE statement
+ *    deparsing as well as functions that are currently only used for deparsing
+ *    ALTER ROLE statements.
+ *
+ * Copyright (c), Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+
+#include "distributed/citus_ruleutils.h"
+#include "distributed/deparser.h"
+#include "lib/stringinfo.h"
+#include "nodes/parsenodes.h"
+#include "utils/builtins.h"
+
+static void AppendAlterRoleStmt(StringInfo buf, AlterRoleStmt *stmt);
+
+
+/*
+ * DeparseAlterRoleStmt builds and returns a string representing of the
+ * AlterRoleStmt for application on a remote server.
+ */
+const char *
+DeparseAlterRoleStmt(AlterRoleStmt *stmt)
+{
+	StringInfoData buf = { 0 };
+	initStringInfo(&buf);
+
+	AppendAlterRoleStmt(&buf, stmt);
+
+	return buf.data;
+}
+
+
+/*
+ * AppendAlterRoleStmt generates the string representation of the
+ * AlterRoleStmt and appends it to the buffer.
+ */
+static void
+AppendAlterRoleStmt(StringInfo buf, AlterRoleStmt *stmt)
+{
+	ListCell *optionCell = NULL;
+	RoleSpec *role = stmt->role;
+
+	/*
+	 * If the role_specification used is CURRENT_USER or SESSION_USER,
+	 * it will be converted to thats roles role name.
+	 */
+	appendStringInfo(buf, "ALTER ROLE %s", RoleSpecString(role));
+
+	foreach(optionCell, stmt->options)
+	{
+		DefElem *option = (DefElem *) lfirst(optionCell);
+
+		if (strcmp(option->defname, "superuser") == 0 && intVal(option->arg))
```

Comment:
I think here you can create a char array and use that instead of having many if elses.

```
char* defNames[7] = {"superuser", "createdb", "createrole", "inherit", "login", "replication", "bypassrls"}
```

And you can iterate the char array to see if `option->defname` is equal to the current element, if so it means that you will append the current element if `intVal(option->arg)` returns true, or you will prepend `NO` if it returns false.

Reply:
> if so it means that you will append the current element

note that few of elements have different defName  than the actual sql string like   `canlogin `   > `LOGIN `    or `isreplication ` -> `REPLICATION`. 

So, that approach might still  require some if statements,  I'd rather keep this, which is long but at least very straightforward

Reply:
Is there a reason why they are different in the first place? I think we can make them the same right? Also we have them hardcoded in two different places, I think ideally we should use a constant variable/define that holds the hard coded strings, so that we use them only from one place and they are not hard coded.

Reply:
that's the output of Postgres parser, we don't have a control on that

Reply:
cant we choose the same output as Postgres?

Reply:
all the strings used in this logic directly related to postgres,

option->defname- > Postgres parser gives us
appendStringInfo(buf, XXX) -> Postgres parser expects us

So, that doesn't seem feasible

Reply:
I see, thanks for the explanation :) 

I think we should at least use a variable instead of having the same hard coded variables in two different places.